Tests/AsyncHTTPClientTests/HTTP2ClientTests.swift

```diff
@@ -187,4 +202,185 @@ class HTTP2ClientTests: XCTestCase {
         // all workers should be running, let's wait for them to finish
         allDone.wait()
     }
+
+    func testUncleanShutdownCancelsExecutingAndQueuedTasks() {
+        let bin = HTTPBin(.http2(compress: false))
+        defer { XCTAssertNoThrow(try bin.shutdown()) }
+        let clientGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer { XCTAssertNoThrow(try clientGroup.syncShutdownGracefully()) }
+        let client = self.makeClientWithActiveHTTP2Connection(to: bin, eventLoopGroupProvider: .shared(clientGroup))
```

Comment:
Why do we need a client with an active connection here? Would love a comment...

Reply:
We need an active connection to guarantee that requests are executed immediately without waiting for connection establishment. Maybe we could also write a test that observes when a connection has been established and only starts shutdown afterwards.

Reply:
would you mind putting this into the test?

Reply:
Done